cypress-tests/cypress/e2e/v2/verifyInfrastructure.cy.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Retry invoking a Cypress request until it succeeds or fail a few times in a row.
+ * @param url - target URL
+ * @param service - name of the tested service
+ * @param limit - maximum attempts before failing
+ * @param delaySeconds - delay between attempts
+ * @returns response body from the request or `null`
+ */
+function retryRequest(
+  url: string,
+  service: string,
+  limit = 6,
+  delaySeconds = 20,
+  retries = 1,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Cypress.Response<any> {
+  if (retries > limit) {
+    const minutes = Math.floor((limit * delaySeconds) / 60);
+    throw new Error(
+      `Backend service "${service}" not responding within ${minutes} minutes at URL ${url}.`,
+    );
+  }
+
+  cy.request({
+    url,
+    failOnStatusCode: false,
+  }).then((resp) => {
+    if (resp.status < 400 && !resp.body.error) return resp.body;
+
+    cy.wait(delaySeconds * 1000).then(() =>
+      retryRequest(url, service, limit, delaySeconds, retries + 1),
+    );
+  });
+  return null;
+}
+
+describe("Verify the infrastructure is ready", () => {
+  it("Can interact with the backend components", () => {
+    retryRequest("api/data/version", "Data services");
+    retryRequest("api/notebooks/version", "Notebooks");
+    retryRequest("api/search/version", "Search");
+    retryRequest("api/auth/login", "Gateway");
+    retryRequest("config.json", "UI client");
+
+    // Data service should return a list of default resopurce pools
+    const dataServiceUrl = "/api/data/resource_pools";
+    cy.request(dataServiceUrl).then((resp) => {
+      if (resp.status >= 400 || !resp.body.length)
+        throw new Error("Data service endpoints not working as expected.");
+    });
+
+    // Notebooks should return a list of servers
+    const notebooksUrl = "/api/notebooks/servers";
+    cy.request(notebooksUrl).then((resp) => {
+      if (resp.status >= 400 || !("servers" in resp.body))
+        throw new Error("Notebook endpoints not working as expected.");
+    });
```

Comment:
```suggestion
```

Same as above you dont have to test the notebooks separately. The are handled by the data service now. And you already check that.